src/ode_solve.jl

```diff
@@ -252,6 +271,37 @@ function inner_loss(phi::ODEPhi{C, T, U}, f, autodiff::Bool, t::AbstractVector,
     sum(abs2, dxdtguess .- fs) / length(t)
 end
 
+"""
+L2 inner loss for DAEProblems
+"""
+
+function inner_loss_DAE end
+
+function inner_loss_DAE(phi::ODEPhi{C, T, U}, f, autodiff::Bool, t::Number, θ,
+                    p) where {C, T, U <: Number}
+    sum(abs2, ode_dfdx(phi, t, θ, autodiff) - f(ode_dfdx(phi, t, θ, autodiff), phi, p, t))
```

Comment:
this doesn't make sense.

Reply:
the output of a dae function is the residual, so there's nothing to subtract

Reply:
Do I just make that line `sum(abs2, ode_dfdx(phi, t, θ, autodiff))` then?

Reply:
No, you need to evaluate f... ` sum(abs2,f(ode_dfdx(phi, t, θ, autodiff), phi(t, θ), p, t)))`